hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/scale/ITestS3AInputStreamPerformance.java

```diff
@@ -467,6 +467,35 @@ public void testRandomIORandomPolicy() throws Throwable {
         0, streamStatistics.getAborted());
   }
 
+  @Test
+  public void testSkip() throws Throwable {
```

Comment:
Nice test to show the functionality. We could have a little more verification that we actually skipped bytes by asserting the content read after skipping "n" bytes. As well as adding seeks/reads in between 2 skips to know that we are skipping from the current position.

Reply:
Make a file using `ContractTestUtils.dataset()`, with the contents as alphabet characters like `ContractTestUtils.dataset(256, 'a', 'z');`. This would give you a file with 256 characters in a loop of alphabet from 'a' to 'z'. Now Open this file, seek some position for eg 100. Then skip 200 bytes. Assert what the number of bytes skipped was, and read the file after skipping. similarly, you can skip after some bytes then read the file and assert that the file read after skipping is what we expect the content to be.